nodeJS/orms/prisma_orm.md

```diff
@@ -0,0 +1,157 @@
+### Introduction
+
+Tired of writing raw SQL queries in your projects? In this lesson, we'll talk about Object Relational Mappers (ORM for short). ORMs are tools that let you manipulate data in your database and are widely used in the software industry. We'll dive deep into one ORM popular in the Node.js landscape: [Prisma ORM](https://www.prisma.io/orm).
+
+### Lesson overview
+
+This section contains a general overview of topics that you will learn in this lesson.
+
+- What ORMs are and why they are used.
+- Introduce Prisma ORM.
+- Describe the features of Prisma ORM.
+
+### Challenges with raw SQLin'
+
+With the couple of projects you've done earlier, you might've run into the tediousness of raw SQL queries. Let's briefly discuss some of these challenges.
+
+#### So much more code
+
+You need a `SELECT` statement? Cool, write a query. Need a `SELECT` statement for a different table? Write another query. Or if your programmer senses are tingling, you could create a utility function for `SELECT`. But now you need a `SELECT` statement for specific columns? You'll have to modify the utility function. What about filters and sorting? You get the point.
+
+Replicate the above for insert queries, and all of its variations. Or you could go in a different direction. Create modules per entity. For example, the books module would have:
+
+```javascript
+class Book {
+  async getBooks(filters) {}
+  async getBookById(id) {}
+  async createBook(data) {}
+  async updateBook(id, data) {}
+  async deleteBook(id) {}
+  async getBookAuthors(id) {}
+  async getBookGenres(id) {}
+  // and so on
+}
+```
+
+Or you could mix and match the two approaches. Create a `Database` class and inherit from it in your entity classes. Or you fancy composition, and just want to deal with pure functions.
+
+Now do this over and over for each of entity, across multiple projects. You get the point. It can be argued that the above is not a bad thing, as you're learning and practicing, and it isn't. It's sufficient for solo projects, but as your codebase grows, and as you work in teams, you'll realize how much an ORM really helps you focus on writing code that's business critical.
```

Comment:
```suggestion
Now do this over and over for each entity, across multiple projects. You get the point. It can be argued that the above is not bad, as you're learning and practicing, and it isn't. It's sufficient for solo projects, but as your codebase grows, and as you work in teams, you'll realize how much an ORM really helps you focus on writing code that's business critical.
```